File: data/backtracking/subset-sum.py
def isSubsetSum(nums: list[int], target: int) -> bool:
    def backtrack(i, current_sum):
        if current_sum == target: return True
        if i >= len(nums) or current_sum > target: return False
        return backtrack(i + 1, current_sum + nums[i]) or backtrack(i + 1, current_sum)

    return backtrack(0, 0)

File: data/backtracking/subset-sum.cpp
#include <vector>

using namespace std;

class SolutionSubsetSum {
    bool backtrack(vector<int>& nums, int target, int i, int currentSum) {
        if (currentSum == target) return true;
        if (i >= nums.size() || currentSum > target) return false;
        return backtrack(nums, target, i + 1, currentSum + nums[i]) || backtrack(nums, target, i + 1, currentSum);
    }
public:
    bool isSubsetSum(vector<int>& nums, int target) {
        return backtrack(nums, target, 0, 0);
    }
};
